### PR TITLE
SNOW-849605: fix datatype misalignment

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-849605

- fix nullable
- fix time type, timestamp type

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Comment by sfc-gh-stan** [comment]
Is this all the misalignment we currently know of?

File: src/snowflake/snowpark/mock/plan.py [hunk 1]
```diff
@@ -622,14 +628,25 @@ def outer_join(base_df):
...
-            result[c].name, result[c].sf_type.datatype, result[c].sf_type.nullable
-        )
-        for c in result.columns
-    ]
+    ret = []
+    for c in result.columns:
+        if isinstance(result[c].sf_type.datatype, NullType):
+            ret.append(
+                Attribute(
+                    result[c].name if result[c].name else "NULL", StringType(), True
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/mock/plan.py:638]: Just curious: Do we support multiple columns of the same name `NULL` from `describe`?

**Review by sfc-gh-sfan — APPROVED** [review]

File: src/snowflake/snowpark/mock/analyzer.py [hunk 2]
```diff
@@ -445,7 +446,17 @@ def unary_expression_extractor(
         expr_to_alias: Dict[str, str],
         parse_local_name=False,
     ) -> str:
-        if isinstance(expr, (Alias, UnresolvedAlias)):
+        if isinstance(expr, Alias):
```
[the remark this answers is not in the document]
Comment (sfc-gh-aling) [reply src/snowflake/snowpark/mock/analyzer.py:463]: I removed that confusing code, it was my mistake.

no hierarchical relationship between the class of Alias and UnresolvedAlias as they are defined:
```pytho
class Alias(UnaryExpression, NamedExpression):
class UnresolvedAlias(UnaryExpression, NamedExpression):
```

**Review by sfc-gh-stan — APPROVED** [review]

File: src/snowflake/snowpark/mock/plan.py (hunk 1 above)
Comment (sfc-gh-aling) [reply src/snowflake/snowpark/mock/plan.py:643]: yes, you will get two columns of name `NULL` by the following scripts:
```python
    df2 = session.create_dataframe([1, 2]).to_df("a").select(lit(None), lit(None))
    assert len(df2.schema.fields) == 2
    assert df2.schema.fields[0].datatype == StringType()
```

**Pull request merged**